backend/src/gpu.py

```diff
@@ -14,6 +14,11 @@
 
 
 class NvidiaHelper:
+    def __new__(cls):
+        if not hasattr(cls, "instance"):
+            cls.instance = super(NvidiaHelper, cls).__new__(cls)
+            return cls.instance
```

Comment:
Man, this is why I hate python. Who in the language design team though that allowing users to change the very nature of class instantiation is a good idea?

Let's please not do this. I had to go on a 10min detour just to figure out wtf this is supposed to be. I don't blame you, because the guides I found really like this black magic, but I don't. 

Let's just have a little helper function `get_nvidia_helper() -> NvidiaHelper | None` that returns `None` when there are no Nvidia GPUs and a cached instance otherwise. This gets rid of the voodoo and also has the advantage that we can't forget to check `nvidia_is_available`.

Reply:
FYI this is just a singleton pattern for a class

Reply:
[This](https://www.baeldung.com/java-singleton#singleton) is how you do a singleton is any other language. `__new__` is just voodoo.